### PR TITLE
Separate infra-sub-properties from infra-params

### PR DESCRIPTION
**Pull request by pasindujw**
**Purpose**
TestGrid infra-params has sub-properties which have been considered as another set of infra-params from one point of the code. This is conceptually wrong and this PR fixes that.

**Approach**
When those sub-properties are needed in the TG flow, TestGrid will fetch them from the database.

**Security checks**
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: common/src/main/java/org/wso2/testgrid/common/infrastructure/InfrastructureCombination.java [hunk 1]
```diff
@@ -47,6 +47,17 @@ public InfrastructureCombination(Set<InfrastructureParameter> parameters) {
...
     }
 
+
+    /**
+     * Initializes an @{@link InfrastructureCombination} object with the given
+     * single infrastructure parameter.
+     *
+     * @param parameter infrastructure parameter.
+     */
+    public InfrastructureCombination(InfrastructureParameter parameter) {
```
Comment (kasunbg) [inline common/src/main/java/org/wso2/testgrid/common/infrastructure/InfrastructureCombination.java]: I'm not sure accepting just one InfrastructureParameter by InfrastructureCombination class is good. How about we change this to accept a var-args instead (`public InfrastructureCombination(InfrastructureParameter parameter...) {
`)?

Alternatively, we can also introduce a method like #addInfrastructureParameter instead.

File: infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java [hunk 2]
```diff
@@ -458,6 +465,19 @@ private Properties getInfraInputs(TestPlan testPlan, Script script) {
...
 
+        //Fetch infra-parms from db and include their sub-props also as properties to be passed.
+        InfrastructureParameterUOW infrastructureParameterUOW = new InfrastructureParameterUOW();
+        Properties infraParamsAsProps =  testPlan.getInfrastructureConfig().getParameters();
+        for (String key: infraParamsAsProps.stringPropertyNames()) {
+            try {
+                infrastructureParameterUOW
+                        .getInfrastructureParameter(infraParamsAsProps.getProperty(key)).get()
+                        .getProcessedSubInfrastructureParameters()
+                        .forEach(infraParam -> props.setProperty(infraParam.getType(), infraParam.getName()));
```
Comment (kasunbg) [inline infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java]: IMHO, infrastructure_parameter table is an abstract detail, and the AWSProvider should not be aware of it. If we couple these two, then it'll be, 1. hard to replace it with something else (for ex. a json file!), 2. we'll have to duplicate above logic in each and every concrete implementations of InfrastructureProvider.
 
So, I think we should always work with the TestPlan object beyond the InfrastructureCombinationProvider class.

WDYT?

File: common/src/main/java/org/wso2/testgrid/common/infrastructure/InfrastructureCombination.java (hunk 1 above)
Comment (pasindujw) [reply common/src/main/java/org/wso2/testgrid/common/infrastructure/InfrastructureCombination.java]: +1

File: infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java (hunk 2 above)
Comment (pasindujw) [reply infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java]: +1

File: dao/src/main/resources/META-INF/persistence.xml [hunk 3]
```diff
@@ -34,7 +34,7 @@
         <properties>
             <property name="javax.persistence.jdbc.url"
                       value="jdbc:mysql://localhost:3306/testgriddb"/>
-            <property name="javax.persistence.jdbc.user" value="root"/>
+            <property name="javax.persistence.jdbc.user" value="root2"/>
```
Comment (kasunbg) [inline dao/src/main/resources/META-INF/persistence.xml]:
```suggestion
            <property name="javax.persistence.jdbc.user" value="root"/>
```

**Pull request merged**